metrics-scm-1.rockspec

```diff
@@ -19,51 +19,10 @@ dependencies = {
-        ['metrics.quantile']                = 'metrics/quantile.lua',
-        ['metrics.http_middleware']         = 'metrics/http_middleware.lua',
-        ['metrics.collectors.shared']       = 'metrics/collectors/shared.lua',
-        ['metrics.collectors.summary']      = 'metrics/collectors/summary.lua',
-        ['metrics.collectors.counter']      = 'metrics/collectors/counter.lua',
-        ['metrics.collectors.gauge']        = 'metrics/collectors/gauge.lua',
-        ['metrics.collectors.histogram']    = 'metrics/collectors/histogram.lua',
-        ['metrics.const']                   = 'metrics/const.lua',
-        ['metrics.plugins.graphite']        = 'metrics/plugins/graphite.lua',
-        ['metrics.plugins.prometheus']      = 'metrics/plugins/prometheus.lua',
-        ['metrics.plugins.json']            = 'metrics/plugins/json.lua',
-        ['metrics.tarantool']               = 'metrics/tarantool.lua',
-        ['metrics.tarantool.fibers']        = 'metrics/tarantool/fibers.lua',
-        ['metrics.tarantool.info']          = 'metrics/tarantool/info.lua',
-        ['metrics.tarantool.memory']        = 'metrics/tarantool/memory.lua',
-        ['metrics.tarantool.memtx']         = 'metrics/tarantool/memtx.lua',
-        ['metrics.tarantool.network']       = 'metrics/tarantool/network.lua',
-        ['metrics.tarantool.operations']    = 'metrics/tarantool/operations.lua',
-        ['metrics.tarantool.replicas']      = 'metrics/tarantool/replicas.lua',
-        ['metrics.tarantool.runtime']       = 'metrics/tarantool/runtime.lua',
-        ['metrics.tarantool.slab']          = 'metrics/tarantool/slab.lua',
-        ['metrics.tarantool.spaces']        = 'metrics/tarantool/spaces.lua',
-        ['metrics.tarantool.system']        = 'metrics/tarantool/system.lua',
-        ['metrics.tarantool.cpu']           = 'metrics/tarantool/cpu.lua',
-        ['metrics.tarantool.event_loop']    = 'metrics/tarantool/event_loop.lua',
-        ['metrics.cartridge.issues']        = 'metrics/cartridge/issues.lua',
-        ['metrics.cartridge.failover']      = 'metrics/cartridge/failover.lua',
-        ['metrics.tarantool.clock']         = 'metrics/tarantool/clock.lua',
-        ['metrics.psutils.cpu']             = 'metrics/psutils/cpu.lua',
-        ['metrics.psutils.psutils_linux']   = 'metrics/psutils/psutils_linux.lua',
-        ['metrics.tarantool.luajit']        = 'metrics/tarantool/luajit.lua',
-        ['metrics.tarantool.vinyl']         = 'metrics/tarantool/vinyl.lua',
-        ['metrics.utils']                   = 'metrics/utils.lua',
-        ['metrics.cfg']                     = 'metrics/cfg.lua',
-        ['metrics.stash']                   = 'metrics/stash.lua',
-        ['metrics.version']                 = 'metrics/version.lua',
-        ['cartridge.roles.metrics']         = 'cartridge/roles/metrics.lua',
-        ['cartridge.health']                = 'cartridge/health.lua',
-    }
+    type = 'cmake',
```

Comment:
> cmake allows more flexible setup, which is needed for automatic override
package build.

What exactly doesn't work with `type = 'builtin'`? I guess that

```lua
modules = {
    ['metrics'] = 'metrics/init.lua',
    ['metrics.api'] = 'metrics/api.lua',
    <...>
    ['override.metrics'] = 'metrics/init.lua',
    ['override.metrics.api'] = 'metrics/api.lua',
    <...>
}
```

would work.

Reply:
Georgy said that it actually doesn't work: each file is installed once (into override/) in the case.

He also shared a way to overcome this problem:

```sh
$ mkdir override
$ ln -s ../metrics override/metrics
```

And in the rockspec:

```lua
modules = {
    ['metrics'] = 'metrics/init.lua',
    ['metrics.api'] = 'metrics/api.lua',
    <...>
    ['override.metrics'] = 'override/metrics/init.lua',
    ['override.metrics.api'] = 'override/metrics/api.lua',
    <...>
}
```

However I agree that it looks like a crutch.

OTOH, it enables the overriding of the built-in module, when one just steps into the repository (it may be convenient at development).

I would say that it is up to the maintener. There are no strict cons or pros.

I would prefer if the commit message will contain more specific motivation behind the builtin -> cmake transition. So we can return here back in a future and understand, say, whether others should do exactly same or it depends.

Reply:
> I would say that it is up to the maintener. There are no strict cons or pros.

I had opened an alternative PR https://github.com/tarantool/metrics/pull/444 . I'll discuss with current maintainer which solution is preferred